ceph: add finalizers to rook-ceph-mon secrets and configmap (#8646)
rook/rook: Documentation/ceph-teardown.md

kubectl api-resources --verbs=list --namespaced -o name \
  | xargs -n 1 kubectl get --show-kind --ignore-not-found -n rook-ceph
```

> ### Remove Critical Resource Finalizers

Reviewer: ```suggestion
### Remove critical resource finalizers
```

Rook adds a finalizer `ceph.rook.io/disaster-protection` to resources critical to the Ceph cluster so that the resources will not be accidentally deleted.

> The operator is responsible for removing the finalizer while deleting the ceph-cluster CR.

Reviewer 2: ```suggestion
The operator is responsible for removing the finalizers when a CephCluster is deleted.
```

> If for some reason the operator is not able to remove the finalizers (ie. the operator is not running anymore), you can remove the finalizers manually with the following commands:

Reviewer 2: ```suggestion
If for some reason the operator is not able to remove the finalizers (i.e., the operator is not running anymore), you can remove the finalizers manually with the following commands:
```

```console

> kubectl patch configmap -n rook-ceph rook-ceph-mon-endpoints --type merge -p '{"metadata":{"finalizers": [null]}}'

Reviewer: ```suggestion
kubectl -n rook-ceph patch configmap rook-ceph-mon-endpoints --type merge -p '{"metadata":{"finalizers": [null]}}'
```

> kubectl patch secrets -n rook-ceph rook-ceph-mon --type merge -p '{"metadata":{"finalizers": [null]}}'

Reviewer: ```suggestion
kubectl -n rook-ceph patch secrets rook-ceph-mon --type merge -p '{"metadata":{"finalizers": [null]}}'
```

kubectl api-resources --verbs=list --namespaced -o name \
  | xargs -n 1 kubectl get --show-kind --ignore-not-found -n rook-ceph
```

> ### Remove Critical Cluster Finalizers

Reviewer 3: ```suggestion
### Remove Critical Resource Finalizers
```

> Added Finalizers `ceph.rook.io/disaster-protection`, to protect the Critical Cluster Information from deletion during disaster scenarios and to recover from it. 

Reviewer 3: ```suggestion
Rook adds a finalizer `ceph.rook.io/disaster-protection` to resources critical to the Ceph cluster so that the resources will not be accidentally deleted.
```

The operator is responsible for removing the finalizer while deleting the ceph-cluster CR.

> If for some reason the operator is not able to remove the finalizer (ie. the operator is not running anymore), you can delete the finalizer manually with the following command:

Reviewer 3: ```suggestion
If for some reason the operator is not able to remove the finalizers (ie. the operator is not running anymore), you can remove the finalizers manually with the following commands:
```

```console
kubectl patch configmap -n rook-ceph rook-ceph-mon-endpoints --type merge -p '{"metadata":{"finalizers": [null]}}'

> 

Reviewer 3: nit: remove blank line
